src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -429,6 +429,9 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
         const hash = tx.hash.toBuffer();
         const index = Quantity.from(i);
 
+        // update the effective gas price of the transaction based off of the
+        // baseFeePerGas of the block that the transaction is actually on
+        tx.updateEffectiveGasPrice(block.header.baseFeePerGas);
```

Comment:
In other places we only do set this if the block header has a `baseFeePerGas`. 

Actually, since `updateEffectiveGasPrice` is already called in the miner itself, why wouldn't the `effectiveGasPrice` be updated already :thinking: